packages/calcite-components/src/utils/dom.ts

```diff
@@ -81,6 +81,19 @@ export function getElementProp(el: Element, attribute: string, fallbackValue: an
   return closest ? closest.getAttribute(attribute) : fallbackValue;
 }
 
+/**
+ * This helper returns the computed width in pixels of a rendered HTMLElement.
+ *
+ * @param {HTMLElement} el An element.
+ * @returns {number} The element's width.
+ */
+export function getElementWidth(el: HTMLElement): number {
+  if (!el) {
+    return;
+  }
+  return Math.round(parseFloat(getComputedStyle(el).inlineSize.replace("px", "")));
```

Comment:
It will return decimal values, so to keep the calculation simpler, I thought it best to round the result to make the math cleaner.  Probably is a negligible difference, could go either way.

![image](https://github.com/Esri/calcite-design-system/assets/821864/d8a69fea-adc4-4227-85f0-09533688ef5a)

Reply:
It will make the calculation that much more accurate I suppose if we don't round the value.  I'll update that.